### PR TITLE
readme.md fixed

### PR DESCRIPTION
**Pull request by denis-bogdanas**

### CODE REVIEW

**Comment by denis-bogdanas** [comment]
@daejunpark Please review when tests pass, to have a fresh gnosis branch. I also added full spec for `encodeTransactionData`.

**Comment by denis-bogdanas** [comment]
jenkins: test this please

**Comment by denis-bogdanas** [comment]
jenkins: test this please

**Comment by denis-bogdanas** [comment]
jenkins: test this please

**Comment by denis-bogdanas** [comment]
jenkins: test this please

**Comment by denis-bogdanas** [comment]
@daejunpark There's something wrong with github. This PR does not show the latest commit. But the tests pass:

https://office.runtimeverification.com/jenkins/job/verified-smart-contracts-pull-request/

Can you review please? Changes are bigger than what description says. Please look at the commit messages.

File: gnosis/gnosis-spec.ini [hunk 1]
```diff
@@ -277,7 +277,10 @@ proxy_storage: _
     andBool G >=Int 10000 //
 
 [encodeTransactionData]
-output: _ => _
+output: _ => #encodeArgs( #bytes(
+                #parseHexWord("0x19") : #parseHexWord("0x1")
+                : #encodeArgs(#bytes32(DOMAIN_SEPARATOR), #bytes32(SAFE_TX_HASH))
+             ))
```
Comment (daejunpark) [inline gnosis/gnosis-spec.ini]: So, the output will be the concatenation of the following bytes?
`bytes32(32)  bytes32(66)  bytes1(0x19)  bytes1(0x1)  bytes32(DOMAIN_SEPARATOR)  bytes32(SAFE_TX_HASH)  bytes30(0)`
and the total size is  160 = 32 + 32 + 1 + 1 + 32 + 32 + 30 ?

In any case, could you put this information as a comment?

File: gnosis/gnosis-spec.ini [hunk 2]
```diff
@@ -277,7 +277,10 @@ proxy_storage: _
...
 statusCode: _ => EVMC_SUCCESS
```
Comment (daejunpark) [inline gnosis/gnosis-spec.ini]: Does `encodeTransactionData` always succeed?

File: gnosis/gnosis-spec.ini [hunk 3]
```diff
@@ -326,12 +329,28 @@ proxy_storage:
     andBool TO =/=Int GAS_TOKEN
     andBool TO =/=Int REFUND_RECEIVER
     andBool GAS_TOKEN =/=Int REFUND_RECEIVER
```
Comment (daejunpark) [inline gnosis/gnosis-spec.ini]: Don't we need to say {TO, GAS_TOKEN, REFUND_RECEIVER} are also distinct to {MASTER_COPY_ID, PROXY_ID, ORIGIN_ID}?

Comment (daejunpark) [reply gnosis/gnosis-spec.ini]: Also, in https://github.com/runtimeverification/verified-smart-contracts/pull/56/files#diff-713cf585e24f9780c47fbb3f16f9de75R301
Don't we need to specify ORIGIN_ID as well?

like:
```
activeAccounts: SetItem(MASTER_COPY_ID) SetItem(PROXY_ID) SetItem(ORIGIN_ID)
```

File: gnosis/gnosis-spec.ini (hunk 2 above)
Comment (denis-bogdanas) [reply gnosis/gnosis-spec.ini]: The specification passes. So yes.

File: gnosis/gnosis-spec.ini (hunk 3 above)
Comment (denis-bogdanas) [reply gnosis/gnosis-spec.ini]: This function doesn't use these arguments as accounts. These 3 inequalities are actually not needed and I deleted them. I guess it's best to keep the specification most general.

File: gnosis/gnosis-spec.ini (hunk 1 above)
Comment (denis-bogdanas) [reply gnosis/gnosis-spec.ini]: done

**Comment by denis-bogdanas** [comment]
jenkins: test this please

File: gnosis/gnosis-spec.ini [hunk 4]
```diff
@@ -296,7 +301,7 @@ callData: #abiCallData("encodeTransactionData", (
             #uint256(NONCE) ))
 callValue: 0
 activeAccounts: SetItem(MASTER_COPY_ID) SetItem(PROXY_ID)
-                SetItem(TO) SetItem(GAS_TOKEN) SetItem(REFUND_RECEIVER) _:Set
+                _:Set
```
Comment (daejunpark) [inline gnosis/gnosis-spec.ini]: In this case, you can just drop the whole `activeAccounts` parameter here, as it is already defined in the `[root]` section.

Comment (daejunpark) [reply gnosis/gnosis-spec.ini]: I mean, we need to specify the `ORIGIN_ID` in the activeAccounts cell. I don't think an inactive account can send a transaction, which you can double-check though.

Comment (denis-bogdanas) [reply gnosis/gnosis-spec.ini]: Done. Waiting the tests. Last time there was a spurious failure on Jenkins. Couldn't reproduce locally.

**Pull request merged**